library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt

```diff
@@ -81,91 +77,94 @@ internal class AuthAPIImpl : AuthAPI {
+                phoneNumber,
+                "",
+                authSignInOptions,
+                { resume(ApiResult.Success(it)) },
+                { resume(ApiResult.Failure(it)) }
+            )
+        }
     }
 
-    override fun confirmSignIn(
-        code: String,
-        handler: AuthRequestHandler
-    ) {
-        Amplify.Auth.confirmSignIn(
-            code,
-            handler::onSuccess,
-            handler::onError,
-        )
+    override suspend fun confirmSignIn(
+        code: String
+    ): ApiResult<AuthSignInResult> {
+        return suspendCancellableCoroutine {
+            Amplify.Auth.confirmSignIn(
+                code,
+                { resume(ApiResult.Success(it)) },
+                { resume(ApiResult.Failure(it)) }
+            )
+        }
     }
 
-    override fun confirmResendCode(
-        code: String,
-        handler: AuthRequestHandler
-    ) {
-        Amplify.Auth.confirmUserAttribute(
-            AuthUserAttributeKey.phoneNumber(),
-            code,
-            handler::onSuccess,
-            handler::onError
-        )
+    override suspend fun confirmResendCode(
+        code: String
+    ): ApiResult<Unit> {
+        return suspendCancellableCoroutine {
+            Amplify.Auth.confirmUserAttribute(
+                AuthUserAttributeKey.phoneNumber(),
+                code,
+                { resume(ApiResult.Success<Unit>(Unit)) },
+                { resume(ApiResult.Failure<Unit>(it)) },
+            )
+        }
     }
 
-    override fun sendCode(
+    override suspend fun sendCode(
         phoneNumber: String,
-        handler: AuthRequestHandler,
-    ) {
-        when (authenticationType) {
-            AuthenticationType.Mobile -> signIn(phoneNumber, handler)
-            is AuthenticationType.Facebook -> sendPhoneCodeByResend(handler)
+    ): ApiResult<Any> {
+        return when (authenticationType) {
+            AuthenticationType.Mobile -> signIn(phoneNumber)
+            is AuthenticationType.Facebook -> sendPhoneCodeByResend()
         }
     }
 
-    override fun signOut(
-        handler: AuthRequestHandler
-    ) {
-        Amplify.Auth.signOut(
-            handler::onSuccess
-        )
+    override suspend fun signOut(): ApiResult<Unit> {
+        return suspendCancellableCoroutine {
+            Amplify.Auth.signOut {
+                resume(ApiResult.Success<Unit>(Unit))
+            }
+        }
     }
 
-    private fun sendPhoneCodeByResend(handler: AuthRequestHandler) {
-        Amplify.Auth.resendUserAttributeConfirmationCode(
-            AuthUserAttributeKey.phoneNumber(),
-            handler::onSuccess,
-            handler::onError
-        )
+    private suspend fun sendPhoneCodeByResend(): ApiResult<AuthCodeDeliveryDetails> {
+        return suspendCancellableCoroutine {
+            Amplify.Auth.resendUserAttributeConfirmationCode(
+                AuthUserAttributeKey.phoneNumber(),
+                { resume(ApiResult.Success(it)) },

```

Comment:
Corresponding to this note of mine:
https://github.com/AnimealProject/animeal_android/pull/209/files#r1229578171
"AuthCodeDeliveryDetails" result is not currently used anywhere in the code, SignUpAndSignInUseCase treats this method as ApiResult\<Unit\>. I suggest to return here 
{ resume(ApiResult.Success(Unit)) },
and add comment above:
\\** Actual return type of onSuccess function here is AuthCodeDeliveryDetails, but currently it's unused *\
In an improbable case if we would actually need to use AuthCodeDeliveryDetailssomewhere in the code, that would be a separate refactoring ticket which will change this structure in the future

Reply:
done